featuremanagement/_models/_allocation.py

```diff
@@ -0,0 +1,193 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+from dataclasses import dataclass
+from ._constants import DEFAULT_WHEN_ENABLED, DEFAULT_WHEN_DISABLED, USER, GROUP, PERCENTILE, SEED
+
+
+class Allocation:
+    """
+    Represents an allocation
+    """
+
+    def __init__(self):
+        self._default_when_enabled = None
+        self._default_when_disabled = None
+        self._user = []
+        self._group = []
+        self._percentile = []
+        self._seed = None
+
+    @classmethod
+    def convert_from_json(cls, json):
+        """
+        Convert a JSON object to Allocation
+
+        :param json: JSON object
+        :type json: dict
+        :return: Allocation
+        :rtype: Allocation
+        """
+        if not json:
+            return None
+        allocation = cls()
+        allocation._default_when_enabled = json.get(DEFAULT_WHEN_ENABLED)
+        allocation._default_when_disabled = json.get(DEFAULT_WHEN_DISABLED)
+        allocation._user = []
+        allocation._group = []
+        allocation._percentile = []
+        if USER in json:
+            allocations = json.get(USER)
+            for user_allocation in allocations:
+                allocation._user.append(UserAllocation(**user_allocation))
+        if GROUP in json:
+            allocations = json.get(GROUP)
+            for group_allocation in allocations:
+                allocation._group.append(GroupAllocation(**group_allocation))
+        if PERCENTILE in json:
+            allocations = json.get(PERCENTILE)
+            for percentile_allocation in allocations:
+                allocation._percentile.append(PercentileAllocation.convert_from_json(percentile_allocation))
+        allocation._seed = json.get(SEED, "")
```

Comment:
Why does seed default to None in the init, but defaults to "" here? I would imagine it should stay None.

Reply:
It should never be none. Fixing the other location.

Reply:
actually, the default should be "allocation\n<feature-name>", checked DotNet.